code/modules/mining/mine_turfs.dm

```diff
@@ -460,6 +460,20 @@ var/global/list/rockTurfEdgeCache
 	N.fullUpdateMineralOverlays()
 	return
 
+/turf/simulated/mineral/attack_hand(mob/living/user)
+	if(check_affinity(user.mind, AFFINITY_EARTH) && user.a_intent == "help")
```

Comment:
it's still snowflake.

think about it: Every mineral wall ```attack_hand()``` checks if you're a witch, when it 99% of the time you wont be)